File: ruby/simple_sample/sample.rb
require 'rubygems'
require 'selenium-webdriver'
require 'browserstack-patch'

BrowserStack.for "ruby"

@repeat = ENV['REPEAT'] || 5

# Input capabilities
driver = Selenium::WebDriver.for :firefox

begin
  driver.navigate.to "http://www.google.com"
  element = driver.find_element(:name, 'q')
  element.send_keys "BrowserStack"
  element.submit
  start = Time.now.to_i
  @repeat.times do
    puts driver.title
  end
  driver.quit
rescue
  driver.quit
end

File: examples/simple_sample/sample.js
var webdriver = require('selenium-webdriver');

require('browserstack-automate').Node();

var repeat = process.env.REPEAT || 5;

var driver = new webdriver.Builder().
  forBrowser('firefox').
  build();

driver.get('http://www.google.com');

for(var i=0; i<repeat; i++){
  driver.findElement(webdriver.By.name('q')).sendKeys('BrowserStack');
}

driver.findElement(webdriver.By.name('btnG')).click();
driver.getTitle().then(function(title) {
  console.log(title);
});

driver.quit();
